1733Sprint/04-Xander/CH3-demo/GameManager.java

```diff
@@ -0,0 +1,197 @@
+import zakmon.*;
+import java.util.*;
+import java.time.*;
+
+public class GameManager {
+  private List<String> speciesList = new ArrayList<>();
+  private Scanner sc = new Scanner(System.in);
+  private LocalTime currentTime = LocalTime.now();
+
+  public static void main(String[] args) {
+    GameManager gm = new GameManager();
+    gm.populateSpeciesList();
+    gm.greeting();
+    gm.selectZakmonText();
+
+    Zakmon playerMon = new Zakmon(gm.selectZakmon());
+    Zakmon cpuMon = new Zakmon(gm.cpuSelectZakmon());
+
+    gm.battleManager(playerMon, cpuMon);
+  }
+
+  private void battleManager(Zakmon playerMon, Zakmon cpuMon) {
+    int turn = 1;
+
+    while (playerMon.getHealthStat() > 0 && cpuMon.getHealthStat() > 0) {
+      battleStatus(playerMon, cpuMon);
+
+      try {
+        if (turn % 2 == 0) {
+          cpuSelectMove(playerMon, cpuMon);
+        } else {
+          playerSelectMove(playerMon, cpuMon);
+        }
+      } catch (Exception e) {
+        System.out.println("Your "+ playerMon.getSpecies() + " does not understand this move and loses their turn");
+        sc.nextLine();
+      }
+      turn++;
+    }
+    winOrLoseMessage(playerMon, cpuMon);
+  }
+
+  private void cpuSelectMove(Zakmon playerMon, Zakmon cpuMon) {
+    Random r = new Random();
+    int move = r.nextInt(3) + 1;
+
+    if (playerMon.isDefending()) {
+      playerMon.removeDefend();
+    }
+
+    if (move == 1) {
+      int damage = cpuMon.attack(playerMon.getDefenseStat());
+      playerMon.takeDamage(damage);
+      if (cpuMon.getSpecies().equals("Rupspie")) {
+        System.out.println(System.lineSeparator() + "The enemy Rupsie crawls towards you inflicting psychological damage for " + damage + " damage" + System.lineSeparator());
+      } else {
+        System.out.println(System.lineSeparator() + "The enemy attacked and dealt " + damage + " damage" + System.lineSeparator());
+      }
+    } else if (move == 2) {
+      cpuMon.defend();
+      System.out.println(System.lineSeparator() + "The opponent defended" + System.lineSeparator());
+    } else if (move == 3) {
+      System.out.println("The enemy " + cpuMon.getSpecies() + " tries to run but it failed.");
+    }
+  }
+
+  private void playerSelectMove(Zakmon playerMon, Zakmon cpuMon) {
+    int input;
+    playerMoveSelectMessage();
+
+    if (cpuMon.isDefending()) {
+      cpuMon.removeDefend();
+    }
+
+    input = sc.nextInt();
+    switch (input) {
+      case 1:
+        int damage = playerMon.attack(cpuMon.getDefenseStat());
+        cpuMon.takeDamage(damage);
+
+        if (playerMon.getSpecies().equals("Rupspie")) {
+          System.out.println(System.lineSeparator() + "Rupsie crawls towards it's opponent inflicting psychological damge for " + damage + " Damage");
+        } else {
+          System.out.println(System.lineSeparator() + "You attacked and dealt " + damage + " Damage");
+        }
+        break;
```

Comment:
You can create a function in the other class and send the species of player or cpu with it as a parameter. That way your code will be cleaner and more moves can be added to handel the different attacks in a function that covers them.